### PR TITLE
Additional AbsolutePath adaptation that got left out of the previous PR (this should be the rest).

### PR DESCRIPTION
**Pull request by abertelrud**
Additional AbsolutePath adaptation that got left out of the previous PR (this should be the rest).  These were part of the initial testing but were accidentally left out when minimizing the diffs before creating the PR.

### CODE REVIEW

**Comment by abertelrud** [comment]
@swift-ci please test

File: Sources/Build/Command.compile(ClangModule).swift [hunk 1]
```diff
@@ -46,13 +47,13 @@ struct ClangModuleBuildMetadata {
...
     /// Path to working directiory.
-    let prefix: String
+    let prefix: AbsolutePath
 
     /// Extra arguments to append to basic compile and link args.
     let otherArgs: [String]
 
     /// Path to build directory for this module.
-    var buildDirectory: String { return Path.join(prefix, "\(module.c99name).build") }
+    var buildDirectory: AbsolutePath { return prefix.appending(module.c99name + ".build") }
```
Comment (ddunbar) [inline Sources/Build/Command.compile(ClangModule).swift]: Another place where `appending(component:)` would be nice...

**Comment by abertelrud** [comment]
The OS X failure seems to be failures in the Swift compiler itself, but the Linux failure seems real.  Working on a fix.

File: Sources/Build/Command.compile(ClangModule).swift (hunk 1 above)
Comment (abertelrud) [inline Sources/Build/Command.compile(ClangModule).swift]: Agreed.  What do you think should be the behavior of trying to append an empty string or a string containing a `/` in that case?  Throwing an error?

Comment (ddunbar) [inline Sources/Build/Command.compile(ClangModule).swift]: I'd probably say it should be a programmatic error to ever call the method with such a name, clients should diagnose that separately (most likely far earlier).

**Comment by abertelrud** [comment]
@swift-ci

**Comment by abertelrud** [comment]
@swift-ci please test

**Comment by abertelrud** [comment]
@swift-ci please test

File: Sources/Build/Command.compile(ClangModule).swift (hunk 1 above)
Comment (abertelrud) [inline Sources/Build/Command.compile(ClangModule).swift]: Makes sense.

Comment (abertelrud) [inline Sources/Build/Command.compile(ClangModule).swift]: Makes sense.  In that case I think we also want to add API to check for validity, returning a standardized and actionable set of errors.  In other words, a client that takes a name and wants to construct an AbsolutePath should not need to get involved with decisions about what makes a name valid.

**Comment by ddunbar** [comment]
```
Compile Swift Module 'Build' (12 sources)
/home/buildnode/jenkins/workspace/swift-package-manager-PR-Linux@2/swiftpm/Sources/Build/Command.link(SwiftModule).swift:74:25: error: cannot convert value of type 'AbsolutePath' to expected argument type 'String'
            args.append(main)
                        ^~~~
```

**Comment by abertelrud** [comment]
Thanks.  I have a fix but meanwhile I am trying to unbreak my Linux install so I can test this on my desktop some more.

**Comment by abertelrud** [comment]
@swift-ci please test

**Comment by abertelrud** [comment]
Here too, the macOS test failure seems unrelated.

**Comment by abertelrud** [comment]
Lots of merge conflicts now.  Updating again.

**Comment by abertelrud** [comment]
Redid the changes based on the latest master.

**Comment by abertelrud** [comment]
@swift-ci please test

**Comment by abertelrud** [comment]
Once again the macOS failure is unrelated.

**Pull request merged**